docs/examples/move/nft_marketplace/sources/rental_extension.move

```diff
@@ -0,0 +1,417 @@
+module nft_marketplace::rental_extension {
+
+    // iota imports
+    use iota::{
+        kiosk::{Kiosk, KioskOwnerCap, purchase},
+        kiosk_extension,
+        bag,
+        transfer_policy::{Self, TransferPolicy, TransferPolicyCap, has_rule},
+        clock::Clock,
+        coin::{Self, Coin},
+        balance::{Self, Balance},
+        iota::IOTA,
+        package::Publisher
+    };
+
+    // rules imports
+    use kiosk::kiosk_lock_rule::Rule as LockRule;
+
+    // === Errors ===
+    const EExtensionNotInstalled: u64 = 0;
+    const ENotOwner: u64 = 1;
+    const ENotEnoughCoins: u64 = 2;
+    const EInvalidKiosk: u64 = 3;
+    const ERentingPeriodNotOver: u64 = 4;
+    const EObjectNotExist: u64 = 5;
+    const ETotalPriceOverflow: u64 = 6;
+
+    // === Constants ===
+    const PERMISSIONS: u128 = 11;
```

Comment:
Add comment "allows place and lock" or rename to "ALLOW_PLACE_AND_LOCK"

Reply:
Done